src/group-configurations/group-configuration-container/GroupConfigurationContainer.test.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+
+import {
+  contentGroupsMock,
+  experimentGroupConfigurationsMock,
+} from '../__mocks__';
+import messages from './messages';
+import GroupConfigurationContainer from '.';
+
+const contentGroup = contentGroupsMock.groups[0];
+const experimentGroup = experimentGroupConfigurationsMock[0];
+const contentGroupWithUsages = contentGroupsMock.groups[1];
+const contentGroupWithOnlyOneUsage = contentGroupsMock.groups[2];
+
+const renderComponent = (props) => render(
+  <IntlProvider locale="en">
+    <GroupConfigurationContainer group={contentGroup} {...props} />
+  </IntlProvider>,
+);
+
+describe('<GroupConfigurationContainer />', () => {
+  it('renders component correctly', () => {
+    const { getByText, getByTestId } = renderComponent();
+    expect(getByText(contentGroup.name)).toBeInTheDocument();
+    expect(
+      getByText(
+        messages.titleId.defaultMessage.replace('{id}', contentGroup.id),
+      ),
+    ).toBeInTheDocument();
+    expect(getByText(messages.notInUse.defaultMessage)).toBeInTheDocument();
+    expect(getByTestId('configuration-card-header-edit')).toBeInTheDocument();
+    expect(getByTestId('configuration-card-header-delete')).toBeInTheDocument();
+  });
+
+  it('expands/collapses the container group content on title click', () => {
+    const {
+      getByText, queryByTestId, getByTestId, queryByText,
+    } = renderComponent();
+    const cardTitle = getByTestId('configuration-card-header__button');
+    fireEvent.click(cardTitle);
```

Comment:
[important] Using `userEvent` is preferable since this tool [more accurately simulates user behavior](https://blog.mimacom.com/react-testing-library-fireevent-vs-userevent/). We try to avoid using `fireEvent` except in special cases

Reply:
Please check all the tests in this pull request and replace `fireEvent` with `userEvent` where possible

Reply:
changed